lib/ASRouter.jsm

```
@@ -327,12 +333,36 @@ class _ASRouter {
     }
   }
 
+  _validPreviewEndpoint(url) {
+    try {
+      const endpoint = new URL(url);
+      if (!WHITELIST_HOSTS[endpoint.host]) {
+        Cu.reportError(`The preview URL host ${endpoint.host} is not in the whitelist`);
+      }
+      return (endpoint.protocol === "https:" && WHITELIST_HOSTS[endpoint.host]);
```

can you also report an error for if the protocol is not https?